src/src/PluginStructs/P130_data_struct.cpp

```diff
@@ -0,0 +1,283 @@
+                log += " sqrI=" + String((unsigned long)(sqrI >>16), HEX) + String((unsigned long)(sqrI & 0x0000FFFF), HEX);
+                log += " sumI=" + String((unsigned long)(sumI >>16), HEX) + String((unsigned long)(sumI & 0x0000FFFF), HEX);
+                addLog(LOG_LEVEL_INFO, log);
+                #endif
+            }
+        }
+        if ( true == pbmWhileReading ) {
+            success = false;
+        } else {
+            adcIrms = sqrt(1. * sumI / nbSample);
+            success = true;
+        }
+    }
+    if ( false == this->writeRegister(P130_ADS1X15_REG_POINTER_CONFIG, configStop) ) {
+        addLog(LOG_LEVEL_INFO, F("readadccontinuous stop reading failed"));
+    }
+    return success;
+}
+
+float_t P130_data_struct::estimatePower(float_t current) {
+    return calVoltage * current;
+}
+
+boolean P130_data_struct::readCurrent(uint8_t canal, float_t& currentValue) {
+    boolean success = false;
+    boolean calculatingRMSSuccess = true;
+    uint16_t registerValue = 0;
+    int16_t adc = 0;
+    uint64_t sumI = 0;
+    float_t adcIrms = .0;
+    uint16_t nbSample = 0;
+    uint16_t durationSample = (this->nbSinus * (1000 / this->currentFreq));
+    float_t voltage = .0;
+    float_t calCurrent =.0;
+    unsigned long endTime = 0;
+    if ( ( 1 != canal ) && ( 2 != canal ) )
+        success = false;
+    else {
+        uint16_t muxConf = P130_ADS1015_MUX_DIFF_01;
+        if ( 1 == canal )
+            muxConf = P130_ADS1015_MUX_DIFF_01;
+        else if ( 2 == canal )
+            muxConf = P130_ADS1015_MUX_DIFF_23;
+        else
+            muxConf = P130_ADS1015_MUX_DIFF_01; //default value, should never happened
+
+        if ( 1 != this->convModeContinuous ) {
+            // start loop
+            endTime = millis() + durationSample; // 2*20ms => 5sinus
+            calculatingRMSSuccess = true;
+            while ( millis() < endTime ) {
+                if ( false == this->readAdcSingleValue(muxConf, adc) ) {
+                    currentValue = 0;
+                    calculatingRMSSuccess = false;
+                    break;
+                } else {
+                    sumI += (adc * adc);
+                    nbSample++;
+                }
+            }
+            if ( true == calculatingRMSSuccess ) {
+                adcIrms = sqrt(sumI / nbSample);
+                voltage = adcIrms * 0.0005;
+                calCurrent = (1==canal)?calCurrent1:calCurrent2;
+                currentValue = voltage * calCurrent;
+                success = true;
+            }
+        } else {
+            adcIrms = 0.;
+            nbSample = 0;
+            success = this->readAdcContinuousRmsValue(muxConf, durationSample, adcIrms, nbSample);
+            voltage = adcIrms * 0.0005;
+            calCurrent = (1==canal)?calCurrent1:calCurrent2;
+            currentValue = voltage * calCurrent;
+        }
+        if (false != this->debug) {
+            String log;

```

Comment:
Please wrap logs like these in 
```c++
#ifndef BUILD_NO_DEBUG

#endif
```
 so the builds meant for as small builds as possible will not have debug code in them.

Also always wrap logs that need string manipulation in checks like `if (logLevelActiveFor(LOG_LEVEL_XXXX)) {`